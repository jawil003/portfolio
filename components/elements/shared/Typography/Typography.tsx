import React, {
  PropsWithChildren,
  useMemo,
} from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/react";
import designSystem from "../../../../styles/designSystem";

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const BaseStyle = (props: any) => css`
  & {
    font-family: ${props?.fontFamily as Props["fontFamily"]};
    text-align: ${props?.align as Props["align"]};
    font-style: ${(props?.italic as Props["italic"])
      ? "italic"
      : "normal"};
    margin: unset;
    text-decoration: ${props?.underline
      ? "underline"
      : "none"};
    max-width: ${props?.maxLength};
    letter-spacing: ${props?.letterSpacing ??
    "normal"};
    word-spacing: ${props?.wordSpacing ??
    "normal"};
    color: ${props?.color};
    word-wrap: ${props?.wrap
      ? "break-word"
      : "unset"};
    padding: unset;
  }
`;

type StyledComponents = {
  weight?: number;
  align?: Props["align"];
  italic?: Props["italic"];
  underline?: Props["underline"];
  maxLength?: Props["maxLength"];
  letterSpacing?: Props["letterSpacing"];
  wordSpacing?: Props["wordSpacing"];
  wrap?: Props["wrap"];
  fontFamily: Props["fontFamily"];
  color?: Props["color"];
};

const H1 = styled.h1<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 300};
    font-size: 6em;
  }
`;

const H2 = styled.h2<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 300};
    font-size: 3.75em;
  }
`;

const H3 = styled.h3<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 400};
    font-size: 3em;
  }
`;

const H4 = styled.h4<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 400};
    font-size: 2.125em;
  }
`;

const H5 = styled.h5<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 400};
    font-size: 1.5em;
  }
`;

const H6 = styled.h6<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 400};
    font-size: 1.25em;
  }
`;

const B1 = styled.p<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 400};
    font-size: 1em;
  }
`;

const B2 = styled.p<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 400};
    font-size: 0.875em;
  }
`;

const S1 = styled.span<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 300};
    font-size: 1em;
  }
`;

const S2 = styled.span<StyledComponents>`
  ${BaseStyle}
  & {
    font-weight: ${({
      weight,
    }: {
      weight?: number;
    }) => weight ?? 300};
    font-size: 0.875em;
  }
`;

export interface Props
  extends PropsWithChildren<unknown> {
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "b1"
    | "b2"
    | "s1"
    | "s2";
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  component?: React.ElementType<any>;
  className?: string;
  align?:
    | "start"
    | "end"
    | "justify"
    | "center";
  bold?:
    | "semi-bold"
    | "bold"
    | "bolder";
  italic?: boolean;
  underline?: boolean;
  maxLength?: string;
  letterSpacing?: string;
  wordSpacing?: string;
  wrap?: boolean;
  color?: string;
  fontFamily?: string;
}

const resolveVariant = (
  variant: Props["variant"],
  component: Props["component"],
  text: Props["children"],
  className: Props["className"],
  bold: Props["bold"],
  align: Props["align"],
  italic: Props["italic"],
  underline: Props["underline"],
  letterSpacing: Props["letterSpacing"],
  wordSpacing: Props["wordSpacing"],
  wrap: Props["wrap"],
  maxLength: Props["maxLength"],
  fontFamily: Props["fontFamily"],
  color?: Props["color"],
) => {
  const weight =
    bold === "bold"
      ? designSystem.font.fontWeight
          .bold
      : bold === "semi-bold"
      ? designSystem.font.fontWeight
          .semiBold
      : bold === "bolder"
      ? designSystem.font.fontWeight
          .bolder
      : undefined;

  switch (variant) {
    case "h1": {
      return (
        <H1
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          letterSpacing={letterSpacing}
          wordSpacing={wordSpacing}
          wrap={wrap}
          fontFamily={fontFamily}
          maxLength={maxLength}
          as={component}
        >
          {text}
        </H1>
      );
    }
    case "h2": {
      return (
        <H2
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          letterSpacing={letterSpacing}
          wordSpacing={wordSpacing}
          wrap={wrap}
          maxLength={maxLength}
          fontFamily={fontFamily}
          as={component}
        >
          {text}
        </H2>
      );
    }
    case "h3": {
      return (
        <H3
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          wordSpacing={wordSpacing}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          fontFamily={fontFamily}
          wrap={wrap}
          as={component}
        >
          {text}
        </H3>
      );
    }
    case "h4": {
      return (
        <H4
          color={color}
          align={align}
          weight={weight}
          underline={underline}
          italic={italic}
          className={className}
          wordSpacing={wordSpacing}
          letterSpacing={letterSpacing}
          fontFamily={fontFamily}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </H4>
      );
    }
    case "h5": {
      return (
        <H5
          color={color}
          align={align}
          italic={italic}
          weight={weight}
          underline={underline}
          className={className}
          wordSpacing={wordSpacing}
          fontFamily={fontFamily}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </H5>
      );
    }

    case "h6": {
      return (
        <H6
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          fontFamily={fontFamily}
          wordSpacing={wordSpacing}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </H6>
      );
    }
    case "b1": {
      return (
        <B1
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          wordSpacing={wordSpacing}
          fontFamily={fontFamily}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </B1>
      );
    }
    case "b2": {
      return (
        <B2
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          wordSpacing={wordSpacing}
          fontFamily={fontFamily}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </B2>
      );
    }
    case "s1": {
      return (
        <S1
          color={color}
          align={align}
          weight={weight}
          italic={italic}
          underline={underline}
          className={className}
          wordSpacing={wordSpacing}
          fontFamily={fontFamily}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </S1>
      );
    }
    case "s2": {
      return (
        <S2
          color={color}
          align={align}
          weight={weight}
          underline={underline}
          italic={italic}
          className={className}
          wordSpacing={wordSpacing}
          fontFamily={fontFamily}
          letterSpacing={letterSpacing}
          maxLength={maxLength}
          wrap={wrap}
          as={component}
        >
          {text}
        </S2>
      );
    }
  }
};

/**
 * An Typography React Component.
 * @author Jannik Will
 * @version 0.1
 */
export const Typography: React.FC<Props> = ({
  variant,
  component,
  children,
  className,
  bold,
  align,
  italic,
  color,
  underline,
  letterSpacing,
  maxLength,
  wordSpacing,
  wrap,
  fontFamily,
}) => {
  return (
    <>
      {useMemo(
        () =>
          resolveVariant(
            variant,
            component,
            children,
            className,
            bold,
            align,
            italic,
            underline,
            letterSpacing,
            wordSpacing,
            wrap,
            maxLength,
            fontFamily,
            color,
          ),
        [
          align,
          bold,
          children,
          className,
          component,
          variant,
          italic,
          color,
          underline,
          letterSpacing,
          wordSpacing,
          maxLength,
          wrap,
          fontFamily,
        ],
      )}
    </>
  );
};

Typography.defaultProps = {
  wrap: false,
  variant: "b1",
  align: "start",
  color:
    designSystem.colors.brand
      .secondaryText,
};
