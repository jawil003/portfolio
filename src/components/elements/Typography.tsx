import React, {
  forwardRef,
  PropsWithChildren,
} from "react";
import { css } from "@emotion/react";
import designSystem from "src/styles/designSystem";

export const typographyClasses = css`
  .bold {
    font-weight: 700;
  }
  .semi-bold {
    font-weight: 600;
  }
  .bolder {
    font-weight: 600;
  }
  .italic {
    font-styles: italic;
  }
  .underline {
    text-decoration: underline;
  }
  .align-left {
    text-align: left;
  }
  .align-center {
    text-align: center;
  }
  .align-right {
    text-align: right;
  }
  .align-justify {
    text-align: justify;
  }
`;

const getFontSizeForVariant = (
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "b1"
    | "b2"
    | "b3"
    | "custom",
) => {
  switch (variant) {
    case "h1": {
      return 5;
    }
    case "h2": {
      return 4;
    }
    case "h3": {
      return 3;
    }
    case "h4": {
      return 2;
    }
    case "h5": {
      return 1.5;
    }
    case "h6": {
      return 1.1;
    }

    case "b2": {
      return 0.8;
    }
    case "b3": {
      return 0.6;
    }
    default: {
      return 1;
    }
  }
};

const generatestyles = (
  fontSize: number,
  color?: string,
  fontFamily?: string,
  align?:
    | "left"
    | "center"
    | "right"
    | "justify",
  bold?:
    | "semi-bold"
    | "bold"
    | "bolder",
  italic?: boolean,
  underline?: boolean,
  maxLength?: string,
  letterSpacing?: number,
  wordSpacing?: number,
  wrap?: boolean,
) => {
  return css`
    & {
      font-size: ${fontSize}rem;
      font-weight: ${bold ===
      "semi-bold"
        ? designSystem.font.fontWeight
            .semiBold
        : bold === "bold"
        ? designSystem.font.fontWeight
            .bold
        : bold === "bolder"
        ? designSystem.font.fontWeight
            .bold
        : fontSize > 1
        ? designSystem.font.fontWeight
            .semiBold
        : designSystem.font.fontWeight
            .light};
      ${color
        ? `color: ${color}`
        : undefined};
      ${fontFamily
        ? `font-family: ${fontFamily}`
        : undefined};
      text-align: ${align};
      font-style: ${italic
        ? "italic"
        : "normal"};
      text-decoration: ${underline
        ? "underline"
        : "normal"};
      max-width: ${maxLength};
      letter-spacing: ${letterSpacing};
      word-spacing: ${wordSpacing};
      word-wrap: ${wrap
        ? "break-word"
        : "normal"};
    }
  `;
};

export interface Props {
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "b1"
    | "b2"
    | "b3"
    | "custom";
  size?: "string";
  color?: string;
  fontFamily?: string;
  align?:
    | "left"
    | "center"
    | "right"
    | "justify";
  bold?:
    | "semi-bold"
    | "bold"
    | "bolder";
  italic?: boolean;
  underline?: boolean;
  inline?: boolean;
  maxLength?: string;
  letterSpacing?: number;
  wordSpacing?: number;
  wrap?: boolean;
  className?: string;
}

/**
 * An Typography React Component.
 * @author Jannik Will
 * @version 0.1
 */
const Typography: React.FC<Props> = forwardRef<
  | HTMLHeadingElement
  | HTMLParagraphElement,
  PropsWithChildren<Props>
>(
  (
    {
      variant,
      color,
      children,
      fontFamily,
      align,
      bold,
      italic,
      underline,
      inline,
      maxLength,
      letterSpacing,
      wordSpacing,
      wrap,
      className,
    },
    ref,
  ) => {
    const styles = generatestyles(
      getFontSizeForVariant(variant),
      color,
      fontFamily,
      align,
      bold,
      italic,
      underline,
      maxLength,
      letterSpacing,
      wordSpacing,
      wrap,
    );
    if (variant?.includes("b")) {
      if (inline)
        return (
          <>
            <span
              className={className}
              ref={ref}
              css={[styles]}
            >
              {children}
            </span>
          </>
        );
      return (
        <>
          <p
            ref={ref}
            className={className}
            css={[styles]}
          >
            {children}
          </p>
        </>
      );
    }
    switch (variant) {
      case "h1": {
        return (
          <>
            <h1
              ref={ref}
              className={className}
              css={[styles]}
            >
              {children}
            </h1>
          </>
        );
      }
      case "h2": {
        return (
          <>
            <h2
              ref={ref}
              className={className}
              css={[styles]}
            >
              {children}
            </h2>
          </>
        );
      }
      case "h3": {
        return (
          <>
            <h3
              className={className}
              ref={ref}
              css={styles}
            >
              {children}
            </h3>
          </>
        );
      }
      case "h4": {
        return (
          <>
            <h4
              className={className}
              ref={ref}
              css={[styles]}
            >
              {children}
            </h4>
          </>
        );
      }
      case "h5": {
        return (
          <h5
            className={className}
            ref={ref}
            css={styles}
          >
            {children}
          </h5>
        );
      }
      default: {
        return (
          <h6
            className={className}
            ref={ref}
            css={[styles]}
          >
            {children}
          </h6>
        );
      }
    }
  },
);

Typography.displayName = "Typography";

Typography.defaultProps = {
  variant: "b1",
  color:
    designSystem.colors.brand
      .secondaryText,
  align: "left",
  italic: false,
  underline: false,
  inline: false,
  wrap: false,
  maxLength: "100%",
};

export default Typography;
