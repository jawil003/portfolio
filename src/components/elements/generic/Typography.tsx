import React, {
  forwardRef,
  PropsWithChildren,
} from "react";
import css from "styled-jsx/css";

export const typographyClasses = css`
  .bold {
    font-weight: 700;
  }
  .semi-bold {
    font-weight: 600;
  }
  .bolder {
    font-weight: 600;
  }
  .italic {
    font-styles: italic;
  }
  .underline {
    text-decoration: underline;
  }
  .align-left {
    text-align: left;
  }
  .align-center {
    text-align: center;
  }
  .align-right {
    text-align: right;
  }
  .align-justify {
    text-align: justify;
  }
`;

const generatestyles = (
  fontSize: number,
  color?: string,
  fontFamily?: string,
  align?:
    | "left"
    | "center"
    | "right"
    | "justify",
  bold?:
    | "semi-bold"
    | "bold"
    | "bolder",
  italic?: boolean,
  underline?: boolean,
  maxLength?: string,
  letterSpacing?: number,
  wordSpacing?: number,
  wrap?: boolean,
) => {
  return css.resolve`
    .root {
      font-size: ${fontSize}rem;
      font-weight: ${bold ===
      "semi-bold"
        ? 600
        : bold === "bold"
        ? 700
        : bold === "bolder"
        ? 900
        : fontSize > 1
        ? 700
        : 400};
      color: ${color};
      font-family: ${fontFamily};
      text-align: ${align};
      font-style: ${italic
        ? "italic"
        : "normal"};
      text-decoration: ${underline
        ? "underline"
        : "normal"};
      max-width: ${maxLength};
      letter-spacing: ${letterSpacing};
      word-spacing: ${wordSpacing};
      word-wrap: ${wrap
        ? "break-word"
        : "normal"};
    }
  `;
};

export interface Props {
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "b1"
    | "b2"
    | "b3"
    | "custom";
  size?: "string";
  color?: string;
  fontFamily?: string;
  align?:
    | "left"
    | "center"
    | "right"
    | "justify";
  bold?:
    | "semi-bold"
    | "bold"
    | "bolder";
  italic?: boolean;
  underline?: boolean;
  inline?: boolean;
  maxLength?: string;
  letterSpacing?: number;
  wordSpacing?: number;
  wrap?: boolean;
}

/**
 * An Typography React Component.
 * @author Jannik Will
 * @version 0.1
 */
const Typography: React.FC<Props> = forwardRef<
  | HTMLHeadingElement
  | HTMLParagraphElement,
  PropsWithChildren<Props>
>(
  (
    {
      variant,
      color,
      children,
      fontFamily,
      align,
      bold,
      italic,
      underline,
      inline,
      maxLength,
      letterSpacing,
      wordSpacing,
      wrap,
    },
    ref,
  ) => {
    const {
      className,
      styles,
    } = generatestyles(
      variant === "b1"
        ? 1
        : variant === "b2"
        ? 0.8
        : 0.6,
      color,
      fontFamily,
      align,
      bold,
      italic,
      underline,
      maxLength,
      letterSpacing,
      wordSpacing,
      wrap,
    );
    if (variant?.includes("b")) {
      if (inline)
        return (
          <span ref={ref}>
            <style jsx>{`
              ${typographyClasses}
              span {
                color: ${color};
                max-width: ${maxLength};
                font-family: ${fontFamily};
                letter-spacing: ${letterSpacing};
                word-spacing: ${wordSpacing};
              }
            `}</style>
            {children}
          </span>
        );
      return (
        <p ref={ref}>
          <style jsx>{`
            ${typographyClasses}
            p {
              color: ${color};
              max-width: ${maxLength};
              font-family: ${fontFamily};
              letter-spacing: ${letterSpacing};
              word-spacing: ${wordSpacing};
            }
          `}</style>
          {children}
        </p>
      );
    }
    switch (variant) {
      case "h1": {
        return (
          <>
            <h1
              ref={ref}
              className={`${className}`}
            >
              {children}
            </h1>
            {styles}
          </>
        );
      }
      case "h2": {
        return (
          <h2
            ref={ref}
            className={`${className} root`}
          >
            {styles}
            {children}
          </h2>
        );
      }
      case "h3": {
        return (
          <>
            <h3
              ref={ref}
              className={`${className} root`}
            >
              {children}
            </h3>
            {styles}
          </>
        );
      }
      case "h4": {
        return (
          <h4
            ref={ref}
            className={`${className} root`}
          >
            {styles}
            {children}
          </h4>
        );
      }
      case "h5": {
        return (
          <h5
            ref={ref}
            className={`${className} root`}
          >
            {styles}
            {children}
          </h5>
        );
      }
      default: {
        return (
          <h6
            ref={ref}
            className={`${className} root`}
          >
            {styles}
            {children}
          </h6>
        );
      }
    }
  },
);

Typography.displayName = "Typography";

Typography.defaultProps = {
  variant: "b1",
  color: "var(--secondaryText)",
  align: "left",
  italic: false,
  underline: false,
  inline: false,
  wrap: false,
  maxLength: "100%",
};

export default Typography;
