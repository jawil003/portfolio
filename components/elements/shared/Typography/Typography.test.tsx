import { Typography } from "./Typography";
import { render } from "@testing-library/react";

describe("Typography", () => {
  test("Create H1 Element", () => {
    const testH1Typography = render(
      <Typography variant="h1" />,
    );
    expect(
      testH1Typography.baseElement,
    ).toMatchSnapshot();
  });
  test("Create H2 Element", () => {
    const testH2Typography = render(
      <Typography variant="h2" />,
    );
    expect(
      testH2Typography.baseElement,
    ).toMatchSnapshot();
  });
  test("Create H3 Element", () => {
    const testH3Typography = render(
      <Typography variant="h3" />,
    );
    expect(
      testH3Typography.baseElement,
    ).toMatchSnapshot();
  });
  test("Create H4 Element", () => {
    const testH4Typography = render(
      <Typography variant="h4" />,
    );
    expect(
      testH4Typography.baseElement,
    ).toMatchSnapshot();
  });
  test("Create H5 Element", () => {
    const testH5Typography = render(
      <Typography variant="h5" />,
    );
    expect(
      testH5Typography.baseElement,
    ).toMatchSnapshot();
  });
  test("Create H6 Element", () => {
    const testH6Typography = render(
      <Typography variant="h6" />,
    );
    expect(
      testH6Typography.baseElement,
    ).toMatchSnapshot();
  });
});
