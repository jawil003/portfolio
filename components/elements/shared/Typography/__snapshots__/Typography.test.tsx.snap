// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Create H1 Element 1`] = `
.emotion-0 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: ;
  letter-spacing: ;
  word-spacing: ;
  color: ;
  word-wrap: unset;
  padding: unset;
}

.emotion-0 {
  font-weight: 300;
  font-size: 6em;
  font-style: normal;
}

<body>
  <div>
    <h1
      class="emotion-0 emotion-1"
    />
  </div>
</body>
`;

exports[`Typography Create H2 Element 1`] = `
.emotion-0 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: ;
  letter-spacing: ;
  word-spacing: ;
  color: ;
  word-wrap: unset;
  padding: unset;
}

.emotion-0 {
  font-weight: 300;
  font-size: 3.75em;
}

<body>
  <div>
    <h2
      class="emotion-0 emotion-1"
    />
  </div>
</body>
`;

exports[`Typography Create H3 Element 1`] = `
.emotion-0 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: ;
  letter-spacing: ;
  word-spacing: ;
  color: ;
  word-wrap: unset;
  padding: unset;
}

.emotion-0 {
  font-weight: 400;
  font-size: 3em;
}

<body>
  <div>
    <h3
      class="emotion-0 emotion-1"
    />
  </div>
</body>
`;

exports[`Typography Create H4 Element 1`] = `
.emotion-0 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: ;
  letter-spacing: ;
  word-spacing: ;
  color: ;
  word-wrap: unset;
  padding: unset;
}

.emotion-0 {
  font-weight: 400;
  font-size: 2.125em;
}

<body>
  <div>
    <h4
      class="emotion-0 emotion-1"
    />
  </div>
</body>
`;

exports[`Typography Create H5 Element 1`] = `
.emotion-0 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: ;
  letter-spacing: ;
  word-spacing: ;
  color: ;
  word-wrap: unset;
  padding: unset;
}

.emotion-0 {
  font-weight: 400;
  font-size: 1.5em;
}

<body>
  <div>
    <h5
      class="emotion-0 emotion-1"
    />
  </div>
</body>
`;

exports[`Typography Create H6 Element 1`] = `
.emotion-0 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: ;
  letter-spacing: ;
  word-spacing: ;
  color: ;
  word-wrap: unset;
  padding: unset;
}

.emotion-0 {
  font-weight: 400;
  font-size: 1.25em;
}

<body>
  <div>
    <h6
      class="emotion-0 emotion-1"
    />
  </div>
</body>
`;
